pkg/promtail/position.go

```diff
@@ -115,12 +122,14 @@ func readPositionsFile(filename string) (map[string]int64, error) {
 	buf, err := ioutil.ReadFile(filepath.Clean(filename))
 	if err != nil {
 		if os.IsNotExist(err) {
-			return map[string]int64{}, nil
+			positions := map[string]int64{}
+			err = writePositionFile(filename, positions)
+			return positions, err
```

Comment:
Why do you want to write a empty file if it doesn't exist?

Reply:
ahh, this was leftover from some very early debugging I was doing, saving the empty file serves no purpose, I will remove this